clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -174,6 +315,28 @@ public boolean isAllPartitions() {
         return data.topics() == ALL_TOPIC_PARTITIONS;
     }
 
+    public boolean isAllPartitionsForGroup(String groupId) {
+        OffsetFetchRequestGroup group = data
+            .groupIds()
+            .stream()
+            .filter(g -> g.groupId().equals(groupId))
+            .collect(toSingleton());
```

Comment:
This is going to throw IllegalStateException if a group is included multiple times in a request. If that is an invalid request, we should throw InvalidRequestException before it gets here to ensure we don't return UNKNOWN_SERVER_ERROR. More likely, it is valid and we should not throw an exception.

Reply:
Not sure I follow - are you saying that if there is a group that is included multiple times in the request the `stream()` here would throw an `IllegalStateException`? Are you saying we should put a `try...catch` statement here to catch any `IllegalStateException`?

Reply:
The helper method `toSingleton()` throws IllegalStateException if the list size is greater than one. If a request contains the same group twice, it can appear twice in the list.

Reply:
Ah gotcha, makes sense. For now I have removed the `toSingleton()` helper method and collected it into a list that gets the first element instead.